src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -448,10 +449,14 @@ export class NodeClientRequest extends ClientRequest {
     this.response.statusCode = status
     this.response.statusMessage = statusText
 
-    if (headers) {
+    // Try extracting the raw headers from the headers instance.
+    // If not possible, fallback to the headers instance as-is.
+    const rawHeaders = getRawFetchHeaders(headers) || headers
```

Comment:
We will try to read the raw headers from the internal `Headers` symbols but if those are not present for any reason, fallback to the `headers` instance as-is.